### Evaluation and development of a bioinformatics workflow for amplicon sequencing with nanopore long read technology

____________________________________________________________

#### Tools, Programs, Resources
**CRISPRfinder program online**
https://crispr.i2bc.paris-saclay.fr/Server/
* CRISPR Database for sequence identification
* New Platform on https://crisprcas.i2bc.paris-saclay.fr/  
* Taxonomy > cellular organisms > Bacteria (superkingdom) > Proteobacteria(phylum > Gammaproteobacteria(class) > Enterboacteriales (order) > Erwinia (genus) > Erwinia amylovora (species)

**Docker Docs**  
https://docs.docker.com/
* Docs for setting up Docker

**ZHAW HPC Access and Info**  
https://wiki.hpc.zhaw.ch/hpcuserwiki/index.php/Welcome_to_the_HPC_User_Wiki
* Offers tutorials on accessing the cluster
* offers examples on paralell processing jobs
**Link wiki:** https://wiki.hpc.zhaw.ch/hpcuserwiki/index.php/Welcome_to_the_HPC_User_Wiki
**HPC Hardware Info:**https://wiki.hpc.zhaw.ch/hpcuserwiki/index.php/Services:Earth_cluster
**Link Account:** https://idm01.hpc.zhaw.ch/ipa/ui/
**User name:** voro
**Password:** (Hint: 2x)

**Erwinia amylovora Microbe Wiki**  
https://crisprcas.i2bc.paris-saclay.fr/
* Some general information ont the erwinia amylovora

**Weblink Nonopore**  
https://nanoporetech.com/nanopore-sequencing-data-analysis#tabs-0=custom  
* Tool offered by nanopore  
* Workflows and pipeline for sequence analysisys

**Nanopore SW Downloads**  
https://community.nanoporetech.com/downloads
* EPI2ME, Guppy, etc.

**Weblinke Bash commands in background**  
https://www.maketecheasier.com/run-bash-commands-background-linux/
* Examples of bash command for pputting jobs into the background
____________________________________________________________

#### Weblinkes with infos

**Weblink for better basecalling algorithms**  
https://nanoporetech.com/about-us/news/new-research-algorithms-yield-accuracy-gains-nanopore-sequencing

**Bonito Basecaller**  
https://github.com/nanoporetech/bonito
____________________________________________________________

#### Tutorials

**Weblink Nanopore Training**  
https://denbi-nanopore-training-course.readthedocs.io/en/latest/index.html
* Tutorial welches einen durch die Analyse von Nonopore raw data begleitet
* Siehe Note "Nanopore Training"

**Unicycler for Assembly in EGSB group**  
https://github.com/rrwick/Unicycler
* Test this one

____________________________________________________________

#### Possible Papers and other resources:  
* https://medium.com/intothegenomics/using-docker-in-genomics-workflows-cd366850b185
* https://bioinformatics-core-shared-training.github.io/docker-4-bioinformatics/
