# Master Tasks:
Here the open and  closed tasks are listed. There is categorization depending on the topic. Closed taks are listed at the end of this file in an unstructured manner.

## Current
* [x] Check Picture integration with Vercel
* [ ] Write Keywords for the abstract
* [ ] Write Keywords for the Introduction
* [ ] Write 2 Parapgraphs in the outlook
* [ ] Define were to place the R Notebooks
* [ ] Remove Backlinks in the Vercel build
* [x] Finalize Vercel Website
	* [x] Clean Text according to Definition
	* [x] Implement important links to the Github Repository
* [ ] Start going trough Moritz Korrekturen
* [ ] Phylogeny Picture description
* [ ] Maybe check the highest abundant Genus/Species Comparison as shown in the Excel
* [ ] Evtl. Put Diversity Indexes into the outlook
* [ ] Think about Alpha Beta Gamma diversity Table
* [ ] Ask Janik for PDF Vercel integration
* [ ] Propose further workflow steps for the CRISPR data
* [ ] Check why Crispr run works with a new container but not one from the local repository
* [ ] Run 0018 Infos
	* Processed as "Temp1"
	* BC05 Zymo Mock Community (https://www.zymoresearch.com/collections/zymobiomics-microbial-community-standards/products/zymobiomics-microbial-community-dna-standard)
	* BC01-04 Erwinia/Panthoe/Xantomonas/Pseudomonas
	* Illumina vs MinION Outlook

## General
* [ ] Send Excel and Visio files to the Github Repository
* [ ] Think about how to present dependencies of the tools for the user

## Presentations
* [ ] Transfer ppt into markdown presentation
* [ ] Change Master Thesis Output to the Vercell Website!

## Github Repos
* [ ] Finish Vercel Setup
	* [ ] Adapt the notes for better online visuals
	* [ ] Adapt build.js files (Replace MINT stuff)

* [ ] Write new tasks
	* [ ] Task verify (Check all tools / check if path exists / check if dataset and dataname in exits)
	* [ ] Task run-wf (run a workflow

* [ ] Create Tool dependencies
	* [ ] List the version with which the setup has been tested
	* [ ] Create a short install guide

## Scripts and Workflows

## Writing
 
## Dataset 16S rRNA
* [x] Process data with RDP, Greengenes and SILVA
* [ ] There is also data available run on the Illumina MiSeq

## Dataset amoA Gene
* [ ] Replicate workflow with centrifuge
* [x] Eventually automate it

## Automated Workflow

## General Workflow and Workpackages
* [ ] Setup GPU basecalling with Guppy

## Tools which should be installed on the cluster
* [ ] qcat
* [ ] Kraken2
* [ ] guppy 4.4

## Dataset Priority
* 16S RNA 1st priority
* amoA 2nd priority
* Crispr 2nd priority

# Done
* [x] Proofread Results
* [x] Send Draft to Nicolas for reviewing
* [x] Write Phylogeny
* [x] Write Outlook subsection
* [x] Write Outlook subsection
* [x] Proof Read Results
* [x] Check Methods formatting
* [x] Evtl. Put the Rarefaction curves into the outlook
* [x] Finish: Theory: Taxonomic Classification with CRISPR Regions
* [x] Create Data Flow Diagram for soil analysis and mention it in the tasks
* [x] Create goal box for CRISPR
* [x] Work on CRISPR processing
* [x] One Section Outlook writing
* [x] Proof Read
* [x] Finalize the Github Repository
* [x] Update Graphs
* [x] Update and Proofread Texts
* [x] Add the R Notebooks
* [x] Finish Coverage Change Section
* [x] Start Writing Rarefaction curves
* [x] Finish writing rarefaction curves (Picture description)
* [x] Write around one page
* [x] Finish proofreading theory
* [x] Microbiome Analysis 5.4 to 5.5
* [x] Add Apendix D with Plots
* [x] Write Phylogeny
* [x] Write Outlook
* [x] Write around one page
* [x] Add Abbreviation table
* [x] Send Theory for proof reading + a an overview of available days
* [x] Start Writing Outlook
* [x] Implement Rarefaction curves
* [x] Clean up WP - 03 and 04
* [x] Write: Taxonomic Classification with CRISPR Regions
* [x] Write Results POND framework
* [x] Write further on workflow comparison
* [x] Finish workflow comparison
* [x] Start Writing Database Comparison
* [x] Finish Database Comparison and add VennDiagram
* [x] Correct Texts
* [x] Create Pavian Graphs
* [x] Add Pavian to methods
* [x] Clean Methods layout
* [x] Create Rarefaction curves for D4 and D5
* [x] Write Rough Structure on the outlook 
* [x] Read the Publication about the 3 databases for discussion
* [x] Export the phylogenetic trees for soil microbiome analysis
* [x] Start Writing on the Soil Microbiome Analysis
* [x] Reexport: Section_5.4_ Read_Frequency_Classification.jpg -> Label y axis is wrong
* [x] Separate Soil Analysis notebook and graphs to new directory
* [x] Think about Rarefaction curves
* [x] Create Rarefaction curves
	* Talk about it but show it in the appendix
* [x] Create VennDiagram
* [x] Create all of the Goal boxes
* [x] Task Init (Create dirs)
* [x] Task Task Help (like: https://github.com/janikvonrotz/dotfiles/blob/master/task) (replace docu in github)
* [x] Task list scripts (same as help but shows available scripts)
* [x] Task list (list scripts and workflows)
* [x] Enable strict line breaks
* [x] Bring all markdown files into one dir
* [x] Bring all pictures into assets
* [x] Change main md file to README.md
* [x] Is there a way to navigate arround the not flexible number of arrays for the workflow -> Geert
* [x] Create a for loop based basecalling for automated subdir detection (no adjustment of amount of arrays)
* [x] reate a workflow script and test it
* [x] Is there a way to navigate arround the not flexible number of arrays for the workflow -> Geert
* [x] Further write on the README file -> Mention Workflows and Scripts
* [x] Do the assessment of hdf5 file (evntually with Nanoplot)
* [x] Give histogram of barcodes (before and after filtering)
* [x] Finish the first automated Workflow Setup
* [x]  Write Task Command for Workflow
* [x]  Start automated Workflow for Dataset 6
* [x] Write and Finish Methods (add uDocker)
* [x] Write Results POND
* [x] Write Results until Polishing
* [x] Finish write assessment of tools
* [x] Process Data for DB comparison
* [x] Create the automated workflow and test it
* [x] Correct Title sizes in the Result Graphs (Workflow Comparison)
* [x] Start Processing of D1 for CRISPR analysis
* [x] Adapt Section_5.2 to 5.3 for the Workflow Comparsion Graphs
* [x] Check why Crispr run works with a new container but not one from the local repositr
* [x] Write Chapter about frameworks
* [x] Write final chapter about frameworks
* [x] Start Theory on CRISPR
* [x] Write about Goals of assessment and discuss them
* [x] Adapt the R Plots for Workflow Comparison
* [x] Create the Plots for the Database Comparison
* [x] Write CRISPR chapter 1
* [x] Joel -> Propose Meeting
* [x] Create Plots for Soil Comparison
* [x] Write Sleep commands for available nodes
* [x] Write Results about Workflow Comparison
* [x] Write Classification Assessment for Workflow Comparison
* [x] Do the assessment of hdf5 file (evntually with Nanoplot)
* [x] Give histogram of barcodes (before and after filtering)
* [x] Finish the first automated Workflow Setup
* [x]  Write Task Command for Workflow
* [x]  Start automated Workflow for Dataset 6
* [x] Create a for loop based basecalling for automated subdir detection (no adjustment of amount of arrays)
* [x] Create a workflow script and test it
* [x] Write and Finish Methods (add uDocker)
* [x] Finish Assessment Results
* [x] Write Results until Polishing
* [x] Finish write assessment of tools
* [x] Process Data for DB comparison
* [x] Create the automated workflow and test it
* [x] Correct Title sizes in the Result Graphs (Workflow Comparison)
* [x] Start Processing of D1 for CRISPR analysis
* [x] Adapt Section_5.2 to 5.3 for the Workflow Comparsion Graphs
* [x] Define User requirements in detail
* [x] Check out installed centrifuge
* [x] Do classification with Kraken2 and Greengenes DB
* [x] Classifiy the filered reads with Kraken2
* [x] Install Krona Tools for Visualization
* [x] Create Scripts for processing Kraken2 results
* [x] Create a krona script and process all D4/5 Datasets
* [x] Create a scirpt for processing the Kraken2 results
* [x] Check Template of Stefans Latex setup
* [x] Checkout uDocker
* [x] Check if only flowcell and kits are used (All other users should select flow cell and kit instead)
* [x] Create a wiki of the obsidian notes on github pages or obsidian publish
* [x] Define the Goals for the Thesis!
* [x] How to check dependencies on the server
* [x] Consolidate structure of the obsidian notes
* [x] Create a compariosn of the Kraken2 Stats
* [x] Solve the taxid problem with the greengenes db
* [x] Reprocess Dataset 4/5 with EPI2ME (2020.04.06 -> Should have Genus Names)
* [x] Filter the EPI2ME and Kraken2 Results for length and quality, then Compare workflows
* [x] Write Generations of Sequencing 
* [x] Add Pictures to the Generatios of Sequencing section
* [x] Write Nanopore Technology section
* [x] Comment the env file (check if command)
* [x] Adjust scripts with the grep command
* [x] Explain Datatransfer in the README
* [x] Install Kraken2
* [x] Start basecalling of Dataset 4
* [x] Start basecalling of Dataset 5
* [x] Check if it is possible to submit a job wenn another is finished
* [x] Start writing
* [x] Start setting up the presentation for the ACLS
* [x] Install Kraken2 and DB
* [x] Start Writing Thesis
* [x] Check guppy performance and start array job
* [x] Transfer publications into Citavi
* [x] Finish Data source Overview
* [x] Create overview of the processed data
* [x] Bring all data to the quality filtered demultiplexed form
* [x] Use Kraken2 for raw demultiplexed dataset 4 and 5
* [x] Check the prename function and finalize array basecalling
* [x] Check Github repository
* [x] Finalize Data description of Dataset 4 and 5
* [x] Array output naming needs to be different!
* [x] Check with geert why the array output is not according to the description
* [x] Does one have to add an additional array because the last points to the slurm job?
* [x] Get a Citavi License key
* [x] Check if there will be container support on the hpc
* [x] Create Presentations with AE and PPT
* [x] Call Geert for the dependency workflow setup
* [x] Test the workflow crash setup
* [x] Local installation of guppy 4.4
* [x] Test Array guppy output not correct!
* [x] Check why arrays are not creating the subfolders
* [x] Create the presentation for the ACSL
* [x] Start basecalling of Dataset 5
* [x] Setup Arrays for faster processing
* [x] Discuss Tasks with Moritz
* [x] kraken / centrifuge already installed on hpc
* [x] Check available literature on amplicon mapping
* [x] Assess the workflows of the data already done 
* [x] Check on how to wait for finished sbatch run
* [x] Work on automated Workflow
* [x] Choose Taxonomy ID Tool for 16S RNA workflow
* [x] Eventually download data for 16S RNA workflow
* [x] Place the workflow publications from the computer on one drive
* [x] Filter and Trim with Nanofilt
* [x] Think about new file structure based on name of dataset for creting the directory and a manifest file
* [x] Finalize Filtering and Trimming excel
* [x] Check if Nanofilt still runs without x-permission on the files
* [x] Write Terminal Output into file (from the qcat application) --> Mail to Geert
* [x] Reintegrate the category table 
* [x] Sent Aufgabenstellung 
* [x] Lukas als dritter Korrektor?
* [x] Outline for Master Thesis and hand it in!
* [x] Setup new working place at IUNR
* [x] Write a description of the data
* [x] Find the corresponding thesis and publication to the data
* [x] Write about source of the data
* [x] Start transferring relevant papers into Citavi
* [x] Check Tool categorization
* [x] Re-annotate categorised Publications
* [x] Learn to write custom bioinformatics Software
* [x] Regex und Rosalind Training!
* [x] Bash Training
* [x] Ask on which node one can tryout running programs
* [x] Create workflow diagrams with the mermaid plugin
* [x] Use content ma_task_10_BashTraining for updating ma_task_09_BashCommands
* [x] Install qcat
* [x] reorder citavi knowledge elements
* [x] Entry bash commands and reorganize
* [x] Change workflow 2 and 3 to Workpackage 1 and 2
* [x] finish demultiplexing of datasets 1-3
* [x] Define quality cut off when data is basecalled -> Visio Workflow
* [x] Create overview of the Workspace setup
* [x] Check assignments to barcodes
* [x] Create a workflow diagram with mermaid plugin and load it to Github
* [x] Create ToDos for the Meeting
* [x] Turn wp_02 into a task
* [x] Read on Nanopore technology