# Open Questions:

* [x] Why does Dataset 1 and 2 have the same flowcell, kits, and sequencing setting?
	* It is another and the informaiton can be found in the datasets
* [ ] Is there a difference between BC and LWB ?
* [ ] From where are the additional Informations on run 0009 (not everything is in the data)
* [x] Why has the run 0007d fastq and fast5 files?
* [x] What is the type of flow cell used ? (possible combinations: https://denbi-nanopore-training-course.readthedocs.io/en/latest/basecalling/basecalling.html)
* [ ] What is the flowcell used for dataset 3

# Dataset 1 (run 0004)
General Description: 
* amoA and Crispr Analysis

Generated by:
* Samuel Wildhaber and Fabio Rezzonico

Thesis 1:
* Investigating the nitrogen cycle using the amoA gene in aquaponics

Amount of data:
* 56.9 gb

Type of file:
* fast5

Basecalled:
* Yes on workstation EGSB-WS06

IUNR ID:
* 0004

Sequencing Parameters:
* 4 hours runtime / Init. Voltage -180mV / Mux scan 0.5h 

Barcoding and Sequencing Kit:
* SQK-PBK004  (mentioned in Samuel Wildhabers Thesis)

Flow Cell:
* FLO-MIN106 (mentioned in Samuel Wildhabers Thesis) / Lot: FAK82993 

Activated Channels begin:
* 1521

Activated Channels end:
* 1309

Expected read size:
* amoA - 550bp / Crr1/Crr2 - 1500-2500 bp

Samples:
* [[dataset_01_SamplesAndInfo]]

### Dataset 1 Workflow done by data owners

amoA Workflow:
* Basecalling: Live with MinKnow
* Demultiplexing: qcat (EPI2ME: FASTQ barcoding workflow: https://community.nanoporetech.com/technical_documents/epi2me-tech-doc/v/metd_5000_v1_revat_29feb2016/barcoding)
	* Expected number of barcodes: 12
* Filtering: Most probably qcat
	* Length: No separation
	* Quality: Q-Score of 7 and higher
* Assembly/Mapping: Canu
	* Error correction with Canu
* Taxonomic ID: BLASTn on BLAST+ 2.3 of the corrected reads



# Dataset 2 (run 0007d)
General Description:
* Crispr analysis from SA

Generated by:
* Simon Carnal and Fabio Rezzonico

Thesis 1:
* Feuerbrand in Apfelwälderökosystemen von Zentralasien

Amount of data:
* 23 gb

Type of file:
* fast5 and fastq

Basecalled:
* Yes on Workstation EGSB-WS06 / Guppy 3.0.7

IUNR ID:
* 0007 / 0007b (0007 and 0007b on 24.10.2019 that failed as only the apters were detected), 0007c / 0007d (run0007d was a restart of run 0007c )

Sequencing Parameters:
* 1 hours runtime / Init. Voltage -180mV / Mux scan 0.5h 

Barcoding and Sequencing Kit:
* SQK-PBK004  (described in the raw data file: MinION_ZHAW_run_0007d.pdf, also mentioned as the successfull run with LWB-Barcodes)

Flow Cell:
* FLO-MIN106 (described in the raw data file: MinION_ZHAW_run_0007d.pdf) / Lot: FAL53437

Activated Channels begin:
* na

Activated Channels end:
* na

Expected read size:
* Crr1/Crr2 - 1500-2500 bp

Samples:[[dataset_02_SamplesAndInfo]]

### Dataset 2 Workflow done by data owners
CRR Workflow:
* Basecalling: Live with MinKnow
* Demultiplexing: EPI2ME (Tool not given)
	* Expected number of barcodes: 12
* Filtering: Nanofilt 
	* Length: Extraction 1520 - 1824 bp / 2128 - 2432 bp / 2432 - 2736 bp
	* Quality: Quality score of > 13
	* ![[amplicon_info_1.png]]
* Assembly/Mapping: Canu
	* reads to contig
* Taxonomic ID: Not given


# Dataset 3
General Description:
*  12 barcoded CRISPR amplicon libraries

Generated by:
* Fabio Rezzonico

Publication:
* ?

Amount of data:
* 13.5 GB

Type of file:
* .fast5

Basecalled:
* Yes on workstation EGSB-WS06

IUNR ID:
* 0009

Sequencing Parameters:
* 4 hours runtime / Init. Voltage -200mV / Mux scan 0.25h 

Barcoding Kit:
* SQK-LSK109 (Ligation) / EXP-PBC001 (Barcode Expansion Kit) (based on the infos of Joel Pothiers Mail)

Flow Cell:
* FLO-MIN106 (Estimated based on use in Dataset 1) / Lot:FAL53759

Activated Channels begin:
* 771

Activated Channels end:
* 653

Expected read size:
* ?

Samples:**[[dataset_03_SamplesAndInfo]]


### Dataset 3 Workflow done by data owners
Not given but most probably similar to Dataset 2

# Dataset 4
General Description:
*  16S Classification of soil samples

Generated by:
* Joel Pothier and Julian Kreiner

Publication:
* na

Amount of data:
* 5.58 Gb

Type of file:
* .fast5

Basecalled and Demultiplexing:
* Yes on workstation EGSB-WS06 / Basecalling and demultiplexing was not performed for run 0014c and 0014d

IUNR ID:
* 00014a-d

Sequencing Parameters:
* 4 hours runtime /Initial voltage was set to -180 mV for runs 0014a, 0014b, -185 mV for 0014c and -190 mV for 0014d

Barcoding Kit:
* SQK-16S024

Flow Cell:
* FLO-FLG001 / AFN211

Activated Channels begin:
* na

Activated Channels end:
* na

Expected read size:* 1000-2000 bp

Samples:
* [[dataset_04_SamplesAndInfo]]

# Dataset 5
General Description:
*  16S Classification of soil samples

Generated by:
* Joel Pothier and Julian Kreiner

Publication:** na

Amount of data:
* 12 Gb 

Type of file:
* .fast5

Basecalled:
* Basecalling and demultiplexing was performed on WS06 with fast basecalling for run 0015a. The other runs were not basecalled/demultiplexed live.

IUNR ID:
* 00015a-d

Sequencing Parameters:
* 12 hours, 6 hours, 12 hours and about 3.75 hours, respectively / Initial voltage was set to -180 mV for run 0015a, -185mV for 0015b, -195 mV for 0015c and -220/-210/-185 mV for 0015d

Barcoding Kit:
* SQK-16S024

Flow Cell:
* FLO-FLG001 / AFN842

Activated Channels begin:
* 77

Activated Channels end:
* na

Expected read size:
* Expected read size:* 1000-2000 bp

Samples:
* [[dataset_05_SamplesAndInfo]]

# Dataset 6
General Description:
*  16S Classification of soil samples and additional ampliconsequencing 

Generated by:
* Joel Pothier and Nicolas

Publication: 
* na

Amount of data:
* ? Gb 

Type of file:
* .fast5

Basecalled:
* Basecalling and demultiplexing was performed on WS06 with fast basecalling for run 0016a. The runs 0016b-c were not basecalled/demultiplexed live.

IUNR ID:
* 00016a-c

Sequencing Parameters:
*  for 1 hour and 12 hours, respectively. Run 0016c was performed on 11.12.2020 for 8 hours after priming again the flow cell with 120 ul of priming mix / Initial voltage was set to -180 mV for both runs and b whereas 190 was used for run 0016c.

Barcoding Kit:
* SQK-16S024

Flow Cell:
* FLO-FLG001 / AFH196

Activated Channels begin:
* 33

Activated Channels end:
* 1

Expected read size:
* 16s rRNA, 1000-2000 bp

Samples:
* [[dataset_06_SamplesAndInfo]]

# Dataset 7
Check Excel File


# Notes
The basecalling was already performed for both runs (performed “live” during the run on our Workstation no. 6, i.e. several days…) and this explains why two folders are present: fast5_fail and fast5_pass. Unfortunately it’s not possible to go back to the ‘real’ raw data (before basecalling) but it can be that it is just the matter of merging these two folders.