### Comment tracking, first review 
* [x] Contents
* [x] 1 Theoretical Background
* [x] 1.1 Nanopore Technology
* [x] 1.2 MinION, real-time DNA and RNA sequencing device
* [x] 1.2.1 MinION and Nanopore Technology, specials use case
* [x] 1.3 Third generation of sequencing technologies
* [x] 1.4 Advantages of the MinION system and sources of error
* [x] 1.5 Basecalling of MinIon reads
* [x] 1.6 Amplicon based long read workflows
* [x] 2.1 Amplicon based long read workflows in practice
* [x] 2.2 Software tools used in amplicon based long read workflows
* [x] 2.3 Workflow proposition
* [x] 3 Discussion and Outlook
* [x] 3.1 Workflow evaluation
* [x] References

### Further ToDos
* [x] **Aspect of longer barcodes for multiplexing. Compare multiplexing in general (MUI MinION). Aspects of combinatorics**
* [x] Grammar correction in general with notes from Joel
* [ ] Shortly write about Oxford Nanopore company (R1)
* [ ] Explain the difference of biological and solid state (R2)
* [ ] I would bring all this up, for example after "mediocre accuracy". (R3)
* [x] Fix Typos: 1.2.1 MinION and Nanopore Technology, specials use cases (R4)
* [x] Make table with advantages and disadvantages
* [x] Check that SGS is spelled out at the first mentioning (R6)
* [x] Include Metabarcoding aspect in discussion (R7)
* [x] Sort the combined table and check grammar in the table (R8)
* [x] If Time add license to table (R9)
* [x] Make References smaller (numbering in text and APA (ZHAW) style in the bibliography)
* [x] Enabling   high-accuracy   long-read   amplicon   sequences   using   unique  molecular   identifiers   with   Nanopore   or   PacBio   sequencing
* [ ] https://www.illumina.com/science/sequencing-method-explorer/kits-and-arrays/umi.html
* [ ] https://www.nature.com/articles/s41598-018-31064-7
* [ ] https://www.nature.com/articles/s41598-018-31064-7
* [ ] https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-4933-1
### Notes
* environmental taxonomy -> does not exist
* Write Fig. # instead of figure number
* Take original legends in pictures form other publicaitons
* Use Ã— isntead of x
* et. al in text in italics
* Use 16S rRNA everywhere
* wet lab
* variables in italics

### Questions
* What is meant by (B1)
* also reduction in chemistry and enzyme required. (B2)
	* [x] No limitations
* iterations in a table? -> It is in the graph (B3)
	* [x] **Summary table of the nanopore gnerations**
* What should be put above? (B4)
	* [x] First and Second Generation Section before third Generation section
* Yes it is metabording, what would speak against it? (B5)
	* [x] fine or just as barcoding
* Errors can occur depending on the type of species and use of amplified or nascent DNA for the training of the basecalling model. Can you aelaborate (B6)
	* [x] Propose training as MA Tasks
* What should be italics? (B7)
	* [x] k-mer
* What is meant? (B8)
	* [x] restructure 
* This table is in my opinion part of a result and not theoretical Background (B8)
	* [x] table 4 and 5 as one. Subcategory basecallers type of model. Referring to the tool table in theory basecalling
* What is meant? (B9)
	* [x] Standardize Table as much as possibles
	* [x] If tool not suitable for 3rd Gen mark it